src/congruence.jl

```diff
@@ -0,0 +1,53 @@
+# Congruent transforms X_A_Xt, Xt_A_X, X_invA_Xt, Xt_invA_X for guaranteed PD return values
+
+for f in (:X_A_Xt, :Xt_A_X)
+    @eval begin
+        function $(f)(A::ScalMat, B::ScalMat)
+            @check_argdims A.dim == B.dim
+            return ScalMat(A.dim, abs2(B.value) * A.value)
+        end
+        $(f)(A::PDiagMat, B::PDiagMat) = PDiagMat(abs2.(B.diag) .* A.diag)
+        $(f)(A::PDiagMat, B::ScalMat) = PDiagMat(abs2(B.value) .* A.diag)
```

Comment:
AFAICT we also have to check that dimensions match in this function:
```suggestion
        function $(f)(A::PDiagMat, B::ScalMat)
            @check_argdims size(A, 1) == B.dim
            return PDiagMat(abs2(B.value) .* A.diag)
        end
```